### PR TITLE
metric store initial integration, work in progress

### PR DESCRIPTION
**Pull request by abaranau**
Here's the branching strategy for metric store change. 
* metric-strore-integrated is an integration branch for the feature
* we'll create PRs and review them against this branch
* so, all code in it is reviewed
* when all done - we'll merge it into develop

In this PR: at least these are missing (hence there are todos in code):

* upgrade metric datasets
* ttl for non-hbase metric datasets
* interpolators
* delete metrics
* metrics discovery
* unit-test framework
* metrics.data package - move
* queue length - overall all unit-tests to pass
* support limit in cube/fact table
* remove + .v2 in dataset names

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    String scope = metricValue.getTags().get(Constants.Metrics.Tag.SCOPE);
+    String measureName = (scope == null ? "system." : scope + ".") + metricValue.getName();
+
+    CubeFact fact = new CubeFact(replaceTagsIfNeeded(metricValue.getTags()),
+                                 toMeasureType(metricValue.getType()), measureName,
+                                 new TimeValue(metricValue.getTimestamp(), metricValue.getValue()));
+    cube.get().add(fact);
+  }
+
+  private Map<String, String> replaceTagsIfNeeded(Map<String, String> original) {
```
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java]: why do we need this ?

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    };
+    this.cube = Suppliers.memoize(new Supplier<Cube>() {
+      @Override
+      public Cube get() {
+        // 1 sec, 1 min, 1 hour and "all time totals"
+        return new DefaultCube(new int[] {1, 60, 3600, Integer.MAX_VALUE}, factTableSupplier, createAggregations());
+      }
+    });
+
+    this.tagMapping = ImmutableMap.of(
```
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java]: comment here about how this tagMapping is used will be useful.

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  }
+
+  private MeasureType toMeasureType(MetricType type) {
+    switch (type) {
+      case COUNTER:
+        return MeasureType.COUNTER;
+      case GAUGE:
+        return MeasureType.GAUGE;
+      default:
+        // should never ha
```
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java]: happen

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java (hunk 1 above)
Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java]: added couple comments. Good point

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java (hunk 2 above)
Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java]: done

**Pull request merged**